ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -91,4 +345,348 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
+          done();
+        });
+      };
+      var addTerm = function() {
+        addTermCalled += 1;
+      };
+      var addTermsComponentRendered = React.addons.TestUtils.
+        renderIntoDocument(
+          <AddTermsComponent
+            vocabularies={vocabularies}
+            repoSlug="repo"
+            addTerm={addTerm}
+            ref={afterMount}
+          />
+        );
+      assert.equal(
+        addTermsComponentRendered.state.errorText,
+        ''
+      );
+      //test error message
+      addTermsComponentRendered.reportError(
+        "Error occurred while adding new term."
+      );
+      assert.equal(
+        addTermsComponentRendered.state.errorText,
+        'Error occurred while adding new term.'
+      );
+    }
+  );
+
+  QUnit.test('Assert that AddVocabulary work properly',
+    function(assert) {
+      assert.ok(AddVocabulary, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var learningResourceTypes = ['course', 'chapter', 'sequential',
+        'vertical', 'html', 'video', 'discussion', 'problem'];
+      var saveVocabularyResponse;
+      var done = assert.async();
+      var updateParent = function(data) {
+        saveVocabularyResponse = data;
+      };
+      var afterMount = function(component) {
+        assert.equal(
+          component.state.name,
+          ''
+        );
+        assert.equal(
+          component.state.description,
+          ''
+        );
+        assert.equal(
+          component.state.vocabularyType,
+          'm'
+        );
+        assert.equal(
+          component.state.learningResourceTypes.length,
+          0
+        );
+        var formNode = assertAddVocabularyCommon(assert, component);
+        waitForAjax(1, function() {
+          assert.equal(
+            component.state.vocabularyType,
+            'm'
+          );
+          assert.equal(
+            component.state.learningResourceTypes.length,
+            0
+          );
+          assert.equal(
+            saveVocabularyResponse.id,
+            vocabulary.id
+          );
+          var inputNodes = React.addons.TestUtils.
+          scryRenderedDOMComponentsWithTag(
+            formNode,
+            'input'
+          );
+          inputNodes[0].value = "TestA";
+          inputNodes[1].value = "TestA";
+          inputNodes[2].checked = true;
+          React.addons.TestUtils.Simulate.change(
+              inputNodes[2],
+              {target: {value: 'course'}}
+            );
+          inputNodes[11].checked = true;
+          React.addons.TestUtils.Simulate.change(
+              inputNodes[11],
+              {target: {value: 'f'}}
+            );
+          React.addons.TestUtils.Simulate.submit(formNode);
+          waitForAjax(1, function() {
+            assert.equal(
+              saveVocabularyResponse.learning_resource_types[0],
```

Comment:
You are just asserting against the mocked AJAX here so the assertion is meaningless. The next assertion is enough to say that the AJAX call executed and `saveVocabularyResponse` was set properly